Use better L1 regularization range for Logistic

JesseLivezey opened the pull request:
Also makes some changes in Poisson for consistency.

Closes #105

pssachdeva commented:
Is it possible to get tests on logistic that examine each side of the lambda max, to verify that it's a suitable bound? This would effectively be a test on `reg_params`, not UoI Logistic.

JesseLivezey commented:
@pssachdeva Accidentally removed the test during rebase. I added it back in now and added tests for `standardize=True` in the base logistic and poisson classes.

pssachdeva commented:
Looks good to me. Feel free to merge.
